**Repeat Part B but use 100 epochs this time for training**

In [1]:
import pandas as pd
import numpy as np

In [2]:
concrete_data = pd.read_csv('https://cocl.us/concrete_data')
concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

In [3]:
from sklearn.model_selection import train_test_split
train_set, test_set = train_test_split(concrete_data,test_size=0.30)
print('train {0}'.format(train_set.shape))
print('test {0}'.format(test_set.shape))

train (721, 9)
test (309, 9)


In [4]:
# need to use train_set and test_set
concrete_data_columns = train_set.columns

predictors_train_set = train_set[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
target_train_set = train_set['Strength'] # Strength column
#test set
predictors_test_set = test_set[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
target_test_set = test_set['Strength'] # Strength column

In [5]:
predictors_norm_train_set = (predictors_train_set - predictors_train_set.mean()) / predictors_train_set.std()
predictors_norm_test_set = (predictors_test_set - predictors_test_set.mean()) / predictors_test_set.std()

predictors_norm_train_set.head()

num_cols = predictors_norm_train_set.shape[1] # number of predictors
print(num_cols)

8


In [6]:
import keras
from keras.models import Sequential
from keras.layers import Dense
# define regression model
def regression_model_one_layer():
    # create model with one hidden layer
    model = Sequential()
    model.add(Dense(10, activation='relu', input_shape=(num_cols,)))
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

Using TensorFlow backend.


In [7]:
# get the list for mse for printing
import statistics 
from sklearn.metrics import mean_squared_error

# build the model
model = regression_model_one_layer()
mse_results=[]
for x in range(50):
    # fit the model
    model.fit(predictors_norm_train_set, target_train_set, epochs=100, verbose=0)
    train_results = model.predict(predictors_norm_test_set)
    mse_results.append(mean_squared_error(target_test_set, train_results))

Instructions for updating:
Colocations handled automatically by placer.
Instructions for updating:
Use tf.cast instead.


In [8]:
print("Mean of the list of mean square errors is % s" % (statistics.mean(mse_results)))
print("Standard deviation of the list of mean square errors is % s" % (statistics.stdev(mse_results)))

Mean of the list of mean square errors is 51.34128068906027
Standard deviation of the list of mean square errors is 18.44913035561989


**How does the mean of the mean squared errors compare to that from Step B?**

| Stat | A | B | C |
| --- | --- | --- | --- |
| mean | 58.447 | 121.984 | 51.341 |
| s.d. | 34.571 | 25.052 | 18.449 |

The mean and the s.d. are lower (better) than the version B